open-ce/conda_env_file_generator.py

```diff
@@ -0,0 +1,93 @@
+"""
+*****************************************************************
+Licensed Materials - Property of IBM
+(C) Copyright IBM Corp. 2020. All Rights Reserved.
+US Government Users Restricted Rights - Use, duplication or
+disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+*****************************************************************
+"""
+
+import os
+import yaml
+import utils
+
+class CondaEnvFileGenerator():
+    """
+    The CondaEnvData class holds all of the information needed to generate a conda
+    environment file that can be used to create a conda environment from the built packages.
+    """
+
+    #pylint: disable=too-many-instance-attributes,too-many-arguments
+    def __init__(self,
+                 python_versions=utils.DEFAULT_PYTHON_VERS,
+                 build_types=utils.DEFAULT_BUILD_TYPES,
+                 channels=None,
+                 output_folder=None,
+                 env_file_prefix=utils.CONDA_ENV_FILENAME_PREFIX,
+                 ):
+        self.python_versions = utils.parse_arg_list(python_versions)
+        self.build_types = utils.parse_arg_list(build_types)
+        self.env_file_prefix = env_file_prefix
+        self.dependency_dict = {}
+        self.channels = []
+        self._initialize_dependency_dict()
+        self._initialize_channels(channels, output_folder)
+
+    def _initialize_dependency_dict(self):
+        for py_version in self.python_versions:
+            for build_type in self.build_types:
+                key = utils.variant_key(py_version, build_type)
+                self.dependency_dict[key] = set()
```

Comment:
This is missing the MPI_TYPE variant. And isn't very scalable.
Instead of a nested for loop here for all variant types, I think you may want to iterate through the possible variants with this function:
https://github.com/open-ce/open-ce/blob/035395fec283c51e7489f18a82eff678e379335c/open-ce/utils.py#L120-L125

Reply:
Yes, I've incorporated MPI_TYPE. I hadn't noticed that this PR got merged and I didn't pull it.